pybamm/models/submodels/interface/sei/sei_growth.py

```diff
@@ -22,28 +22,55 @@ class SEIGrowth(BaseModel):
     **Extends:** :class:`pybamm.sei.BaseModel`
     """
 
-    def __init__(self, param, reaction_loc, options=None):
-        super().__init__(param, options=options)
+    def __init__(self, param, reaction_loc, options=None, cracks=False):
+        super().__init__(param, options=options, cracks=cracks)
         self.reaction_loc = reaction_loc
 
     def get_fundamental_variables(self):
-        if self.reaction_loc == "x-average":
-            L_inner_av = pybamm.standard_variables.L_inner_av
-            L_outer_av = pybamm.standard_variables.L_outer_av
-            L_inner = pybamm.PrimaryBroadcast(L_inner_av, "negative electrode")
-            L_outer = pybamm.PrimaryBroadcast(L_outer_av, "negative electrode")
-        elif self.reaction_loc == "full electrode":
-            L_inner = pybamm.standard_variables.L_inner
-            L_outer = pybamm.standard_variables.L_outer
-        elif self.reaction_loc == "interface":
-            L_inner = pybamm.standard_variables.L_inner_interface
-            L_outer = pybamm.standard_variables.L_outer_interface
+        if self.reaction == "SEI on cracks":
+            if self.reaction_loc == "x-average":
+                L_inner_av = pybamm.Variable(
+                    "X-averaged inner SEI on cracks thickness",
+                    domain="current collector",
+                )
+                L_inner = pybamm.PrimaryBroadcast(
+                    L_inner_av, "negative electrode"
+                )
+                L_outer_av = pybamm.Variable(
+                    "X-averaged outer SEI on cracks thickness",
+                    domain="current collector",
+                )
+                L_outer = pybamm.PrimaryBroadcast(
+                    L_outer_av, "negative electrode"
+                )
+            elif self.reaction_loc == "full electrode":
+                L_inner = pybamm.Variable(
+                    "Inner SEI on cracks thickness",
+                    domain="negative electrode",
+                    auxiliary_domains={"secondary": "current collector"},
+                )
+                L_outer = pybamm.Variable(
+                    "Outer SEI on cracks thickness",
+                    domain="negative electrode",
+                    auxiliary_domains={"secondary": "current collector"},
+                )
+        else:
```

Comment:
Does this case include standard SEI only, or a bunch of different cases? If it is standard SEI, would it be better to do an `elif` and then keep the `else` to throw an error if the argument is weird?

Reply:
What error should I throw?